pkg/controller/registry/resolver/installabletypes.go

```diff
@@ -77,12 +77,5 @@ func (r SubscriptionInstallable) Constraints() []solver.Constraint {
 func (r *SubscriptionInstallable) AddDependency(dependencies []solver.Identifier) {
 	if len(dependencies) > 0 {
 		r.constraints = append(r.constraints, solver.Dependency(dependencies...))
-		// "dependencies" may contain bundles that are not mutually exclusive
-		// currently, the constraint types:
-		// - packagename+version range
-		// - provided gvk
-		// - ANDs of either of the above
-		// all happen to be mutually exclusive, but this won't always be the case
-		r.constraints = append(r.constraints, solver.AtMost(1, dependencies...))
```

Comment:
so what's the reasoning behind this change if we want at most one solution based on the dependencies? If now we are adding several dependencies we expect to only get one answer in a deterministic way, without explicitly specifying it in the SAT solver?

Reply:
Constraints that prevent duplicate package entries and duplicate GVKs are constructed here:
https://github.com/operator-framework/operator-lifecycle-manager/blob/3b086a72f96ff5e5ad44d2b7337cafd60c7965b0/pkg/controller/registry/resolver/resolver.go#L115
That makes the constraint, "no more than one of the operators that satisfies a subscription's constraints may be installed" redundant.

This one was causing problems because the varargs slice wasn't guaranteed to contain only unique entries, which produced constraints like `AtMost(1, {A, A})` that could never cause anything to be installed.